hbase-client/src/main/java/org/apache/hadoop/hbase/client/TableDescriptor.java

```diff
@@ -275,11 +275,11 @@
 
   /**
    * Check if normalization enable flag of the table is true. If flag is false
-   * then no region normalizer won't attempt to normalize this table.
+   * then region normalizer won't attempt to normalize this table.
    *
-   * @return true if region normalization is enabled for this table
+   * @return value of NORMALIZATION_ENABLED key for this table if present else return null
    */
-  boolean isNormalizationEnabled();
+  Boolean isNormalizationEnabled();
```

Comment:
I think null handling could cause compatibility issue and hence might not make this eligible to land on branch-2 and branch-1 minor releases. @Apache9 @ndimiduk Wondering if it is good idea to deprecate this one and add new API that returns an enum with values: {TRUE, FALSE, NOT_PRESENT_IN_TD} to clearly migrate from here to new one?
This is to maintain source compatibility and with new API, we can make decision of whether to use site config or TD property value and then it can go to minor releases on branch-2.

Reply:
@virajjasani - Not sure how using enum will help in maintaining compatibility. Yes it will surely improve code redabolity but It will face the same problem. Am i missing something here?

Reply:
For IA.Public APIs, we can add new API but should not alter/remove existing ones without going through deprecation cycle. I am proposing something like this:
```
@Deprecated
boolean isNormalizationEnabled();

TDNormalizer isNormalizationEnabled();
```

Going from boolean to Boolean introduces null handling that could break source compatibility and also the behaviour of API changes: instead of returning `false` if normalizer is not defined in TD, now we are returning `null`. Hence, this breaks behaviour of existing IA.Public API.

Reply:
Another way is to move the decision logic to the method in Table descriptor. i personally don't like that but it make sense since whole idea is to set the default value of table descript from false to something that can be set using hbase-default (hbase-site)

Reply:
@virajjasani - Have pushed new changes. I believe i have taken care of compatibility issue as well as the behavior as similar to old one by default.  Please have a look.